flang/lib/Lower/PFTBuilder.cpp

```diff
@@ -221,6 +221,14 @@ class PFTBuilder {
   /// Initialize a new construct and make it the builder's focus.
   template <typename A>
   bool enterConstructOrDirective(const A &construct) {
+    // First handle top level directive case
+    if constexpr (std::is_same_v<parser::CompilerDirective, A>) {
+      if (parentVariantStack.size() == 1) {
+        addUnit(lower::pft::CompilerDirectiveUnit(construct,
+                                                  parentVariantStack.back()));
```

Comment:
This seems like an odd spot to be adding something to the list of units. Seems like it should go in a Pre/Post method like the other units. (Would have to use some template majik sauce to select directives, but...)

Reply:
Good point, I have moved this into a Pre method.